toolchain/check/context.h

```diff
@@ -84,64 +84,65 @@ class Context {
 
   // Returns the current scope, if it is of the specified kind. Otherwise,
   // returns nullopt.
-  template <typename NodeT>
-  auto GetCurrentScopeAs() -> std::optional<NodeT> {
-    auto current_scope_node_id = current_scope().scope_node_id;
-    if (!current_scope_node_id.is_valid()) {
+  template <typename InstT>
+  auto GetCurrentScopeAs() -> std::optional<InstT> {
+    SemIR::InstId current_scope_inst_id = current_scope().scope_inst_id;
```

Comment:
Preserve `auto` for now:
```suggestion
    auto current_scope_inst_id = current_scope().scope_inst_id;
```